Sources/Datadog/RUM/AutoInstrumentation/Views/UIKitHierarchyInspection/UIKitHierarchyInspector.swift

```diff
@@ -0,0 +1,69 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import UIKit
+
+/// Inspects the view controllers hierarchy of the current window and finds
+/// the `UIViewController` which is currently displayed to the user.
+internal protocol UIKitHierarchyInspectorType {
+    /// The top-most `UIViewController` currently displayed to the user.
+    func topViewController() -> UIViewController?
+}
+
+internal struct UIKitHierarchyInspector: UIKitHierarchyInspectorType {
+    private let rootViewControllerProvider: () -> UIViewController?
+
+    init(
+        rootViewControllerProvider: @escaping () -> UIViewController? = {
+            UIApplication.shared.keyWindow?.rootViewController
```

Comment:
Indeed 👍. Changed, however, I did `.first()` instead of `.last()` as it sounds a better default to me - in multi window environment, it's more likely that the "first" window is the one we are interested (vs some other key window created later in the app lifecycle). Anyway, it's a limitation for now.

Reply:
i don't know if there can be multiple key windows, so maybe `first` / `last` makes no difference.?
but if it does:
> The windows in the array are ordered from back to front by window level; thus, the last window in the array is on top of all other app windows

i think we need `last` as it receives user actions. wdyt?

Reply:
All windows can receive the user action 🙂, e.g. given this "floating bubble" in messaging app, its window will just receive the action first (and if not captured, it will be forwarded to the underlying window with messaging UI). My motivation for `first` as default is: it's more likely that the "main" window will be created first (messaging UI), and the accessory window ("floating bubble") will appear later.

Reply:
let's say the user opens the bubble and taps on a button
this will look like _"a tap action on Messaging view"_ (instead of _"on Bubble view"_) in Explorer
is this correct and the expected result?

Reply:
This won't be "a tap action on Messaging view", as `UIKit` will not forward the `UIEvent` to the underlying window. No action will be recorded on the `firsts` window if something happens on `last` - and vice-versa.

Reply:
ah okay, the caught action will be ignored as its identity doesn't match 👌